DK2 SDK v2.9.0 update

apullin opened the pull request:
Updates DK2 project for Nordic SDK v2.9.0 

Also resolves many build warnings throughout the project.

skywinder commented:
Hi @apullin, thanks for the fix.

I can’t run it on my DK2 — it just blinks red after the build. Did it work on yours? Did you test it? If so, let’s troubleshoot what’s causing the problem.

Here’s my setup — please double-check yours, and also send the U2F file so I can compare it with mine. Feel free to send it on Discord.

In addition, please update the docs and README — include how to build with the new config and what settings need to be installed.

![CleanShot 2025-05-05 at 17 24 12@2x](https://github.com/user-attachments/assets/6a01868b-6f9a-4d60-894c-7fe8895d4473)

apullin commented:
@skywinder This DID work for me when I tried it. But I also had to go through a length process to recover my DK2 due to some failure to flash. It's possible it's an incompatibility with the "SoftDevice", since I updated to whatever the latest is.

- will dig into it and figure out the issue
- TODO update readme

skywinder commented:
OK, thank you! I’m also curious how you were able to recover your Omi, because I know at least three people with the same problem.

It happened to me too, but I just connected the debugger and updated the bootloader because it stopped responding to anything.

If you can share your steps or process to rescue it, that would be super helpful — especially for people in Discord who also have dead Omi DK2s.

skywinder commented:
Also, I would like to ask if you will make updates. Let’s split it into two pull requests : fixing empty spaces and actual changes. I agree that it should be changed, and all these trailing spaces are very annoying. 

- But otherwise, it's quite hard to review and find where there are just trailing spaces and where there are the changes. 

It also shows that there is some unused variable, so probably we will remove it as well. What is the first step? Let’s make it work with minimal changes. 

Thanks for the help.

skywinder commented:
just a reference to dead DK issue in disocord, in case if we will find the rescue:
https://discord.com/channels/1192313062041067520/1364435262956114061

apullin commented:
> If you can share your steps or process to rescue it, that would be super helpful — especially for people in Discord who also have dead Omi DK2s.

In my case, it was only partly "dead".
https://discord.com/channels/1192313062041067520/1231903583717425153/1366623165958193163

The symptoms were:
- red LED would blink
- mass storage would enumerate
- looking at "INFO_UF2.txt", it showed: `SoftDevice: none` at the end

To resolve this, I checked out:
https://github.com/adafruit/Adafruit_nRF52_Bootloader

and did all the appropriate setup. AFAIK you'll have to have an `arm-none-eabi-gcc` toolchain installed locally and added to your `PATH`.

and did:
`make BOARD=feather_nrf52840_express SERIAL=/dev/ttyACM0 flash-dfu`
which did visibly work - DFU detected the device and flahed it.

thereafter, I did:
`make BOARD=feather_nrf52840_express flash-sd`

After that, the "INFO_UF2.txt" did show a version for "SoftDevice".

I _actually_ don't recall if that alone made it so that copying a UF2 in made it work.  
At some point, I deleted the `CURRENT.UF2` in the USB mass storage drive that enumerates when plugged in (which doesn't really make sense), and then dropped in the `zephyr.uf2`, and it then flahed, worked, booted up into the Omi FW.

A final note is that I observed TWO red LED states, one with a red LED "breathing" quickly, another with it going slow.
Unusure what these two states correspond to.

Info on what the chip can actually do wrt to DFU mode and UART bootloader mode is spread out all over Nordic's site ... horrible that they don't just publish a Reference Manual.
If there's not even a red LED showing, maybe the bootloader itself is nuked. But it should still be possible to get it into DFU mode, if the right pin is accessible

beastoin commented:
> > If you can share your steps or process to rescue it, that would be super helpful — especially for people in Discord who also have dead Omi DK2s.
> 
> In my case, it was only partly "dead". https://discord.com/channels/1192313062041067520/1231903583717425153/1366623165958193163
> 
> The symptoms were:
> 
>     * red LED would blink
> 
>     * mass storage would enumerate
> 
>     * looking at "INFO_UF2.txt", it showed: `SoftDevice: none` at the end
> 
> 
> To resolve this, I checked out: https://github.com/adafruit/Adafruit_nRF52_Bootloader
> 
> and did all the appropriate setup. AFAIK you'll have to have an `arm-none-eabi-gcc` toolchain installed locally and added to your `PATH`.
> 
> and did: `make BOARD=feather_nrf52840_express SERIAL=/dev/ttyACM0 flash-dfu` which did visibly work - DFU detected the device and flahed it.
> 
> thereafter, I did: `make BOARD=feather_nrf52840_express flash-sd`
> 
> After that, the "INFO_UF2.txt" did show a version for "SoftDevice".
> 
> I _actually_ don't recall if that alone made it so that copying a UF2 in made it work. At some point, I deleted the `CURRENT.UF2` in the USB mass storage drive that enumerates when plugged in (which doesn't really make sense), and then dropped in the `zephyr.uf2`, and it then flahed, worked, booted up into the Omi FW.
> 
> A final note is that I observed TWO red LED states, one with a red LED "breathing" quickly, another with it going slow. Unusure what these two states correspond to.
> 
> Info on what the chip can actually do wrt to DFU mode and UART bootloader mode is spread out all over Nordic's site ... horrible that they don't just publish a Reference Manual. If there's not even a red LED showing, maybe the bootloader itself is nuked. But it should still be possible to get it into DFU mode, if the right pin is accessible

interesting.

beastoin commented:
1/ PDM change is obviously, but why do we need these changes man ?

<img width="595" alt="Screenshot 2025-05-26 at 14 14 10" src="https://github.com/user-attachments/assets/d438d907-0f9e-49e1-9d59-5ea8bc35de6a" />
<img width="1038" alt="Screenshot 2025-05-26 at 14 13 46" src="https://github.com/user-attachments/assets/d9c8d80f-6e7c-40a1-b52e-5197ee0e4caf" />
<img width="344" alt="Screenshot 2025-05-26 at 14 13 32" src="https://github.com/user-attachments/assets/89723d27-24a0-4205-9689-3c239d08aee4" />

@apullin

beastoin commented:
closed due to no updates > 3 days. feel free to reopen it any time man.
